tests/test_integration.py

```diff
@@ -0,0 +1,113 @@
+import pytest
+import numpy as np
+from numpy import testing as npt
+import pandas.util.testing as pdt
+
+from ixmp import Platform
+from message_ix import Scenario
+
+from message_ix.testing import (
+    make_dantzig,
+    models,
+    TS_DF,
+    TS_DF_CLEARED,
+    TS_DF_SHIFT
+)
+
+
+def test_run_clone(tmpdir):
+    # this test is designed to cover the full functionality of the GAMS API
+    # - initialize a new ixmp platform instance
+    # - create a new scenario based on Dantzigs tutorial transport model
+    # - solve the model and read back the solution from the output
+    # - perform tests on the objective value and the timeseries data
+    mp = Platform(tmpdir, dbtype='HSQLDB')
+    scen = make_dantzig(mp, solve=True)
+    assert np.isclose(scen.var('OBJ')['lvl'], 153.675)
+    pdt.assert_frame_equal(scen.timeseries(iamc=True), TS_DF)
+
+    # cloning with `keep_solution=True` keeps all timeseries and the solution
+    # (same behaviour as `ixmp.Scenario`)
+    scen2 = scen.clone(keep_solution=True)
+    assert np.isclose(scen2.var('OBJ')['lvl'], 153.675)
+    pdt.assert_frame_equal(scen2.timeseries(iamc=True), TS_DF)
+
+    # cloning with `keep_solution=False` drops the solution and only keeps
+    # timeseries set as `meta=True` or prior to the first model year
+    # (DIFFERENT behaviour from `ixmp.Scenario`)
+    scen3 = scen.clone(keep_solution=False)
+    assert np.isnan(scen3.var('OBJ')['lvl'])
+    pdt.assert_frame_equal(scen3.timeseries(iamc=True), TS_DF_CLEARED)
+
+
+def test_run_remove_solution(test_mp):
+    # create a new instance of the transport problem and solve it
+    scen = make_dantzig(test_mp, solve=True)
+    assert np.isclose(scen.var('OBJ')['lvl'], 153.675)
+
+    # check that re-solving the model will raise an error if a solution exists
+    pytest.raises(ValueError, scen.solve)
+
+    # check that removing solution with a first-model-year arg raises an error
+    # (DIFFERENT behaviour from `ixmp.Scenario`)
+    pytest.raises(TypeError, scen.remove_solution, first_model_year=1964)
+
+    # check that removing solution does not delete timeseries data
+    # before first model year (DIFFERENT behaviour from `ixmp.Scenario`)
+    scen.remove_solution()
+    pdt.assert_frame_equal(scen.timeseries(iamc=True), TS_DF_CLEARED)
+
+
+def test_shift_first_model_year(test_mp):
+    scen = make_dantzig(test_mp, solve=True, multi_year=True)
+    clone = scen.clone(shift_first_model_year=1964)
+
+    # check that solution and timeseries in new model horizon have been removed
+    assert np.isnan(clone.var('OBJ')['lvl'])
+    pdt.assert_frame_equal(clone.timeseries(iamc=True), TS_DF_SHIFT)
+    # check that the variable `ACT` is now the parameter `historical_activity`
+    assert not clone.par('historical_activity').empty
```

Comment:
should probably assert that this is empty before the clone (otherwise not clear if the test is working)

Reply:
done